spec/contracts/queries/projects/project_queries/shared_contract_examples.rb

```diff
@@ -55,7 +55,51 @@
     context "if the user is not the current user" do
       let(:query_user) { build_stubbed(:user) }
 
-      it_behaves_like "contract is invalid", base: :error_unauthorized
+      it_behaves_like "contract is invalid", base: :can_only_be_modified_by_owner
+    end
+
+    context "if the list is public and the editing user has the permission" do
+      let(:query_user) { build_stubbed(:user) }
+
+      before do
+        query.change_by_system do
+          query.public = true
+        end
```

Comment:
Is `change_by_system` really needed, seems not necessary in this shared spec context

Reply:
Yes, it is needed, because according to the contract, the `public` attribute is not writable except for the `PublishContract`. The same is true for the `user` attribute. That's why we need to set it this way, otherwise the contract will fail because we are trying to write to a field that is not writable.

Reply:
But why not `query.update_attribute!(:public, true)` or alike?

Reply:
I think I got what you mean after checking the whole contract inheritance chain